libpyvinyl/BaseData.py

```diff
@@ -0,0 +1,165 @@
+""" :module BaseData: Module hosts the BaseData class."""
+from abc import abstractmethod
+from libpyvinyl.AbstractBaseClass import AbstractBaseClass
+
+
+class BaseData(AbstractBaseClass):
+    # It represents a python object
+    """The abstract wrapper data class. It's designed in a way so that its inheritance is an API of a
+    kind of data (e.g. wavefront, sample, and diffraction) rather than that of a certain data format."""
+    @abstractmethod
+    def __init__(self,
+                 key,
+                 data_dict=None,
+                 filename=None,
+                 file_format_class=None,
+                 file_format_kwargs=None):
+        self.__key = key
+        excepted_data = {}
+        excepted_data['data1'] = 1
+        excepted_data['data2'] = 2
+        self.__expected_data = excepted_data
+        self.__ioformats = {}
+        self._add_ioformat(openPMDFormat)
+        self._add_ioformat(singfelFormat)
```

Comment:
I don't understand why data1 and data2 as well as singfel format is part of the BaseData class? Maybe this is temporary and just as an example?

Reply:
Hi @mads-bertelsen, the `expected_data` defines what data should the `DataClass` hold and the `__ioformats` defines what format the `DataClass` supports. It's just a placeholder here to let the developer knows what should be included in a `DataClass`. As you can see, the `__init__` is marked as `@abstractmethod`, which means the developer of the inherent class needs to write their own `__init__`.

I would suggest starting from this scatch, where the usage is shown: [libpyvinyl/testBaseData.py](https://github.com/PaNOSC-ViNYL/libpyvinyl/pull/32/files#diff-683f29a1c26c3c41124bf469c5f264dd3ce7a7b815300ba4bbc5eafd70904351)

@CFGrote @shervin86 @mads-bertelsen If you have a better suggestion on how to write the `__init__` of the `DataBase`, please let me know.